src/dragonball/src/device_manager/mod.rs

```diff
@@ -591,14 +591,25 @@ impl DeviceManager {
             .map_err(|_| StartMicroVmError::EventFd)?;
 
         info!(self.logger, "init console path: {:?}", com1_sock_path);
-        if let Some(path) = com1_sock_path {
-            if let Some(legacy_manager) = self.legacy_manager.as_ref() {
-                let com1 = legacy_manager.get_com1_serial();
-                self.con_manager
-                    .create_socket_console(com1, path)
-                    .map_err(StartMicroVmError::DeviceManager)?;
+        match com1_sock_path {
+            Some(path) => {
+                if path == "None" {
+                    info!(
+                        _ctx.logger(),
+                        "This token is used to create stdio console. (means not to create sock console)"
+                    );
+                } else if let Some(legacy_manager) = self.legacy_manager.as_ref() {
+                    let com1 = legacy_manager.get_com1_serial();
+                    self.con_manager
+                        .create_socket_console(com1, path)
+                        .map_err(StartMicroVmError::DeviceManager)?;
+                    return Ok(());
+                }
             }
-        } else if let Some(legacy_manager) = self.legacy_manager.as_ref() {
+            None => {}
```

Comment:
If you do nothing when the `com1_sock_path` is `None`, why not use `if let`?
e.g.
```
if let Some(path) = com1_sock_path {
     if path == "None" {
    .......
} else if ........ {
   .....
}

}
```

Reply:
In `src/api/v1/vmm_action.rs`:

```rust
        let serial_path = match machine_config.serial_path {
            Some(value) => value,
            None => {
                if config.serial_path.is_none() {
                    String::from("/run/dragonball/") + &vm_id + "_com1"
                } else {
                    // Safe to unwrap() because we have checked it has a value.
                    config.serial_path.as_ref().unwrap().clone()
                }
            }
        };
```

This code snippet always assigns a default value to serial_path, namely the `com1_sock_path`. I couldn't figure out whether the default value is required in other places. So in the CLI parser, I set a special string value "None" to make a distinction.

Reply:
I am some kind of confused with the `if let` statement. Since the `com1_sock_path` can not be `None`, and we only need to consider the special string value `"None"`. If I am supposed to do this?

```rust 
if let some("None") == com1_sock_path {
                    info!(
                        ctx.logger(),
                        "This token is used to create stdio console. (means not to create sock console)"
                    );
                }
```

Reply:
```rust
        if let some(path) = com1_sock_path {
            // The token "None" is used to create stdio console. (means not to create sock console)"
            if path != "None" {
                let com1 = legacy_manager.get_com1_serial();
                self.con_manager
                    .create_socket_console(com1, path)
                    .map_err(StartMicroVmError::DeviceManager)?;
                return Ok(());
            }
        }
```

Reply:
Yes! You use `if let` correctly, enjoy Rust:)

Again, please help to see string `"None"` could be optimized.

Reply:
Should I remove the default value of `serial_path` when `serial_path` is `None`? Or I just change another intuitive string value for it?